src/main/java/baron/managers/TaskManager.java

```diff
@@ -0,0 +1,168 @@
+            // command and value which would not be appropriate here since it includes a list() function too
+            if (command.equals(TodoDao.NAME)) {
+                return addTodo(input);
+            } else if (command.equals(DeadlineDao.NAME)) {
+                return addDeadline(input);
+            } else if (command.equals(EventDao.NAME)) {
+                return addEvent(input);
+            } else if (command.equals(Commands.LIST.getCommand())) {
+                return UiManager.list(this.tasks);
+            } else if (command.equals(Commands.MARK.getCommand())) {
+                return mark(input, true);
+            } else if (command.equals(Commands.UNMARK.getCommand())) {
+                return mark(input, false);
+            } else if (command.equals(Commands.DELETE.getCommand())) {
+                return delete(input);
+            } else if (command.equals(Commands.BYE.getCommand())) {
+                // Exits application
+                System.exit(0);
+            } else if (command.equals(Commands.FIND.getCommand())) {
+                return this.find(input);
+            } else {
+                return "Command not recognized";
+            }
+        } catch (Exception ex) {
+            System.out.println(ex.getMessage());
+            return ex.getMessage();
+        }
+        return "Something went wrong, please check the logs for details";
+    }
+
+    protected static String getCommand(String input) {
+        return input.split(" ")[0];
+    }
+
+    private String addTodo(String input) {
+        Todo todo = TodoDao.getFrom(input);
+        TaskDao.add(TodoDao.NAME, todo);
+        return this.add(todo);
+    }
+
+    private String addDeadline(String input) {
+        Deadline deadline = DeadlineDao.getFrom(input);
+        DeadlineDao.add(DeadlineDao.NAME, deadline);
+        return this.add(deadline);
+    }
+
+    private String addEvent(String input) {
+        Event event = EventDao.getFrom(input);
+        EventDao.add(EventDao.NAME, event);
+        return this.add(event);
+    }
+
+    private String mark(String input, boolean isDone) {
+        int taskIndex = Integer.parseInt(StringUtils.getValueOfCommand(input, Commands.MARK.getCommand(), null)) - 1;
+        Task task = this.get(taskIndex);
+        TaskType type = getTaskType(task.toString());
+        task = TaskDao.mark(task.getId(), type.getCommand(), task, isDone);
+        return UiManager.mark(task, isDone);
+    }
+
+    protected String delete(String input) {
+        int i = Integer.parseInt(StringUtils.getValueOfCommand(input, Commands.DELETE.getCommand(), null)) - 1;
+        Task task = this.tasks.remove(i);
+        TaskType type = getTaskType(task.toString());
+        TaskDao.delete(type.getCommand(), task.getId());
+        return UiManager.delete(task, this.tasks.size());
+    }
+
+    /**
+     * Prints out a list of tasks that match the search term (fuzzy search)
+     *
+     * @param input user input
```

Comment:
Perhaps we could be consistent in starting the param descriptors with a capital letter to enhance readability.